src/editors/playgroundController.ts

```diff
@@ -231,11 +274,36 @@ export default class PlaygroundController {
     content: string | undefined
   ): Promise<boolean> {
     try {
-      const document = await vscode.workspace.openTextDocument({
-        language: 'mongodb',
-        content,
+      // The MacOS default folder for saving files is a read-only root (/) directory,
+      // therefore we explicitly specify the workspace folder path
+      // or OS temp directory if a user has not opened workspaces.
+      const workspaceFolder = vscode.workspace.workspaceFolders?.[0];
+      const filePath = workspaceFolder?.uri.fsPath || os.homedir();
```

Comment:
It was `tmpdir`, but Rhys suggested using `homedir` instead. Forgot about the comment, will update it!

Reply:
It does not create a physical file only a reference to a location. Rhys had a point in the design doc thread that since this path is being picked up by vscode to suggest a location where to save the file, the home dir might be more useful rather than the OS temp dir.

Reply:
Cool, SGTM then :+1: